std/path.d

```
@@ -3883,11 +3886,16 @@ string expandTilde(string inputPath) nothrow
                         break;
                     }
 
-                    if (errno != ERANGE)
+                    if (errno != ERANGE &&
+                        // On FreeBSD and OSX, errno can be left at 0 instead of set to ERANGE
+                        errno != 0)
                         onOutOfMemoryError();
 
                     // extra_memory isn't large enough
-                    extra_memory_size *= 2;
+                    import core.checkedint : mulu;
+                    bool overflow;
+                    extra_memory_size = mulu(extra_memory_size, 2, overflow);
+                    if (overflow) assert(0);
```

Well, you could throw an exception, or you could throw a static error singleton as Andrei has proposed before. Both come with their own problems.